rosbag2_transport/src/rosbag2_transport/topic_filter.cpp

```diff
@@ -170,6 +170,12 @@ bool TopicFilter::take_topic(
       }
     }
 
+    if (!record_options_.exclude_topic_types.empty() &&
+      topic_type_in_list(topic_type, record_options_.exclude_topic_types))
+    {
+      return false;
+    }
+
```

Comment:
```suggestion
    if (topic_type_in_list(topic_type, record_options_.exclude_topic_types))
    {
      return false;
    }

```
Don't need to check for an empty vector. The `topic_type_in_list(..)` efficiently handles the use case if the list is empty.